### PR TITLE
feat(route_handler-bpp): consider lane-ego angle diff in lane following module

### LINKED ISSUE #2125: behavior_path_planner generating invalid path orientation
### Checklist

- [X] I've read the [contribution guidelines](https://github.com/autowarefoundation/autoware/blob/main/CONTRIBUTING.md).
- [X] I've searched other issues and no duplicate issues were found.
- [X] I've agreed with the maintainers that I can plan this task.

### Description

When the vehicle passes into the other lane for a reason such as trying to swerve an obstacle while driving in its own lane, an invalid path occurs and the vehicle stops. In order to avoid such situations, it should be checked whether the path created by `behavior_path_planner` is valid.

https://user-images.githubusercontent.com/32412808/196936788-21019595-e3e0-4d36-99d8-94d39b7a964c.mp4

### Expected behavior

It is necessary to check that the path occurred by the `behavior_path_planner` is valid and that the ego can drive. So that invalid routes are prevented from being occurred. The updated path must be generated as a path that the ego can drive.

### Actual behavior

The yaw angle between the path and the vehicle's self pose is not controlled. So that invalid paths can be occurred and vehicle stops.

### Steps to reproduce

1. Set the `rear_drive/front_radius_ratio` and `rear_drive/rear_radius_ratio` values to 2.0 in `obstacle_avoidance_planner.param.yaml` file.
2. Set `threshold_distance_object_is_on_center` to 0.001 in` avoidance.param.yaml`.
3. Set `expand_drivable_area` to true, `left_bound_offset` to 1.9 in `lane_following.param.yaml`
4. Download isuzu_vehicle_launch in [here]( https://drive.google.com/drive/folders/1bkxamrkKsbP3J5339GVf5yKhJL0iii4l)
5. Download [test.zip](https://github.com/autowarefoundation/autoware.universe/files/9836542/test.zip) 
6. run `ros2 launch scenario_test_runner scenario_test_runner.launch.py sensor_model:=sample_sensor_kit vehicle_model:=isuzu_vehicle \ scenario:=/scenario/path/case1.yaml \  architecture_type:=awf/universe launch_rviz:=false launch_autoware:=true`

### PR DESCRIPTION
**Pull request by mehmetdogru**
Signed-off-by: Mehmet Dogru <42mehmetdogru42@gmail.com>

closes #2125 

- [x] :exclamation:  https://github.com/autowarefoundation/autoware_common/pull/148 **must be merged first**
- [x] :exclamation: https://github.com/autowarefoundation/autoware_launch/pull/250 **must be merged first**

## Tests performed

- ~~Please check out also:~~
    - ~~https://github.com/autowarefoundation/autoware_common/pull/148~~ 
    - ~~https://github.com/autowarefoundation/autoware_launch/pull/250~~
- Make sure you use a large vehicle like isuzu bus
- Any map with narrow turns for a bus is testable
- Make sure you expand the drivable area:

```xml
   # Dynamic expansion by projecting the ego footprint along the path
    dynamic_expansion:
      enabled: true
.
.
.
      expansion:
        method: lanelet # method used to expand the drivable area. Either 'lanelet' or 'polygon'.
                         # 'lanelet': add lanelets overlapped by the ego footprints
                         # 'polygon': add polygons built around sections of the ego footprint that go out of the drivable area
```
#### Before the PR (issue):

https://user-images.githubusercontent.com/48479081/224922073-39c61828-bc56-43eb-a316-716376280d13.mp4

#### After PR (fix/feat):

https://user-images.githubusercontent.com/48479081/224922227-50333421-63e4-4152-b9ea-3d5d7a9f1b5f.mp4

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by mehmetdogru** [comment]
friendly ping to review

**Comment by kenji-miyake** [comment]
@mehmetdogru cc @xmfcx Since there are too many reviewers, I think it's good to assign someone.

**Comment by maxime-clem** [comment]
@mehmetdogru @kenji-miyake I will review this PR.

**Comment by yukkysaito** [comment]
cc @TakaHoribe @takayuki5168

**Review by maxime-clem — APPROVED** [review]
The PR looks good. I could not consistently reproduce the issue on the `main` branch, but it appears to be fixed with this PR.
I ran scenarios with this PR and there is no problem (https://evaluation.tier4.jp/evaluation/reports/1c358525-5dc8-50f1-8b87-45a9fbed281a?project_id=prd_jt, private TIER IV link).

File: planning/route_handler/src/route_handler.cpp
[the remark this answers is not in the document]
Comment (mehmetdogru) [reply planning/route_handler/src/route_handler.cpp:1]: Thanks, updated copyright dates.

**Review by purewater0901 — APPROVED** [review]
Seems the owner of the code has to approve this PR, so I give the approval

**Pull request merged**